keras_cv/ops/box_matcher.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Argmax-based box matching"
+
+from typing import List
+from typing import Tuple
+
+import tensorflow as tf
+
+
+class ArgmaxBoxMatcher:
+    """Box matching logic based on argmax of highest value (iou).
+
+    This class computes matches from a similarity matrix. Each row will be
+    matched to at least one column, the matched result can either be positive
+     / negative, or simply ignored depending on the setting.
+
+    The settings include `thresholds` and `indicators`, for example if:
+    1) thresholds=[negative_threshold, positive_threshold], and
+       indicators=[negative_value, ignore_value, positive_value]: the rows will
+       be assigned to positive_value if its argmax result is above
```

Comment:
Isn't this it's `max` result, not its `argmax`?

related: is `ArgmaxBoxMatcher` the right name for this class? It seems like, while the `argmax` is used to compute the result, the thresholding is based on the max value, not its index.

(I may be totally off base here as OD is new to me -- just curious)

Reply:
It's argmax result for the first return